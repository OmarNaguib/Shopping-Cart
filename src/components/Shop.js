import Cards from "./Cards";

export default function Shop({ items, addToCart }) {
  return (
    <div className="shop">
      <Cards items={items} addToCart={addToCart}></Cards>
    </div>
  );
}
