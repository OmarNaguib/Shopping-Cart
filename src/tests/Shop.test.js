import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Shop from "../components/Shop";

const items = [
  {
    src: "/static/media/1table_JOOLA J15 INSIDE TABLE TENNIS TABLE_599.00.dd9ef9f4bbc1398f6cc7.jpg",
    name: "JOOLA J15 INSIDE TABLE TENNIS TABLE",
    price: "599.00",
    inCart: 0,
  },
];

describe("Shop page", () => {
  it("renders with one item", () => {
    const { container } = render(<Shop items={items} addToCart={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
