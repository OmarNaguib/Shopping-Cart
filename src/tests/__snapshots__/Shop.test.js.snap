// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop page renders with one item 1`] = `
<div>
  <div
    class="shop"
  >
    <div
      class="cards"
    >
      <div
        class="card"
      >
        <img
          alt=""
          src="/static/media/1table_JOOLA J15 INSIDE TABLE TENNIS TABLE_599.00.dd9ef9f4bbc1398f6cc7.jpg"
        />
        <div
          class="name"
        >
          JOOLA J15 INSIDE TABLE TENNIS TABLE
        </div>
        <div
          class="price"
        >
          Â£
          599.00
        </div>
        <button>
          Add to Cart
        </button>
        not
      </div>
    </div>
  </div>
</div>
`;
